src/js/track.js

```diff
@@ -0,0 +1,83 @@
+var _ = require('underscore');
+var d3 = require('d3');
+var $ = require('jquery');
+var D3SVGCellRenderer = require('./d3_svg_cell_renderer');
+var utils = require('./utils');
+
+module.exports = {};
+
+var defaultTrackConfig = {
+	label: 'Gene',
+	id_member: 'sample',
+	cell_height: 20,
+	track_height: 20,
+	track_padding: 2.5,
+}; 
+
+function Track(name, oncoprint, data, config) {
+	var self = this;
+	self.name = name;
+	self.config = $.extend({}, defaultTrackConfig, config || {}); // inherit from default
+	
+	self.oncoprint = oncoprint;
+	self.data = data;
+
+	if (self.oncoprint.config.render === 'table') {
+		self.renderer = new TrackTableRenderer(self, new D3SVGCellRenderer(self));
+	}
+
+	self.getLabel = function() {
+		// TODO: label decorations
+		return self.config.label;
+	};
+	
+	self.getDatumIds = function(sort_cmp) {
```

Comment:
So my only experience with OOP is from working with Java. My understanding from this experience is that:

* a getter is a method that returns data which is otherwise completely encapsulated within an object. If it has side-effects, then the name `getMyField` is somewhat misleading because it is both getting and also doing other stuff.
* I think that on line 39 `self.data` is overwritten. This is what a setter does.